app/upgrade_cudos.go

```diff
@@ -585,76 +592,121 @@ func parseGenesisAccounts(jsonData map[string]interface{}, contractAccountMap *O
+func parseGenesisDelegations(validators *OrderedMap[string, *ValidatorInfo], contracts *OrderedMap[string, *ContractInfo], cudosCfg *CudosMergeConfig) (*OrderedMap[string, *OrderedMap[string, sdk.Int]], *OrderedMap[string, *OrderedMap[string, sdk.Int]], error) {
 	// Handle delegations
 	delegatedBalanceMap := NewOrderedMap[string, *OrderedMap[string, sdk.Int]]()
+	unbondingDelegatedBalanceMap := NewOrderedMap[string, *OrderedMap[string, sdk.Int]]()
+
 	for i := range validators.Iterate() {
 		validatorOperatorAddress, validator := i.Key, i.Value
 
-		for j := range validator.delegations.Iterate() {
+		for j := range validator.Delegations.Iterate() {
 			delegatorAddress, delegation := j.Key, j.Value
 
 			resolvedDelegatorAddress, err := resolveIfContractAddressWithFallback(delegatorAddress, contracts, cudosCfg)
 			if err != nil {
-				return nil, err
+				return nil, nil, err
 			}
 
 			currentValidatorInfo := validators.MustGet(validatorOperatorAddress)
-			delegatorTokens := currentValidatorInfo.TokensFromShares(delegation.shares).TruncateInt()
+			delegatorTokens := currentValidatorInfo.TokensFromShares(delegation.Shares).TruncateInt()
 
 			if delegatorTokens.IsZero() {
 				// This happens when number of shares is less than 1
 				continue
 			}
 
 			// Subtract balance from bonded or not-bonded pool
-			if currentValidatorInfo.status == BondedStatus {
+			if currentValidatorInfo.Status == BondedStatus {
 
 				// Store delegation to delegated map
 				resolvedDelegatorMap, _ := delegatedBalanceMap.GetOrSetDefault(resolvedDelegatorAddress, NewOrderedMap[string, sdk.Int]())
 				resolvedDelegator, _ := resolvedDelegatorMap.GetOrSetDefault(validatorOperatorAddress, sdk.NewInt(0))
 				resolvedDelegatorMap.Set(validatorOperatorAddress, resolvedDelegator.Add(delegatorTokens))
 				delegatedBalanceMap.Set(resolvedDelegatorAddress, resolvedDelegatorMap)
+			} else {
+				// Store delegation to delegated map
+				resolvedDelegatorMap, _ := unbondingDelegatedBalanceMap.GetOrSetDefault(resolvedDelegatorAddress, NewOrderedMap[string, sdk.Int]())
+				resolvedDelegator, _ := resolvedDelegatorMap.GetOrSetDefault(validatorOperatorAddress, sdk.NewInt(0))
+				resolvedDelegatorMap.Set(validatorOperatorAddress, resolvedDelegator.Add(delegatorTokens))
+				unbondingDelegatedBalanceMap.Set(resolvedDelegatorAddress, resolvedDelegatorMap)
+			}
+		}
+	}
+
+	return delegatedBalanceMap, unbondingDelegatedBalanceMap, nil
+}
+
+func parseGenesisUnbondingDelegations(validators *OrderedMap[string, *ValidatorInfo], contracts *OrderedMap[string, *ContractInfo], cudosCfg *CudosMergeConfig) (*OrderedMap[string, *OrderedMap[string, sdk.Int]], error) {
+	// Handle delegations
+	unbondedDelegatedBalanceMap := NewOrderedMap[string, *OrderedMap[string, sdk.Int]]()
```

Comment:
The name of this variable is a bit confusing, given the name of this function & what this function does.
I woudl suggest to rename this variable to `unbondingDelegatedBalanceMap`.
```suggestion
	unbondingDelegatedBalanceMap := NewOrderedMap[string, *OrderedMap[string, sdk.Int]]()
```